# 1. nedēļa: ML problēmas definēšana

Šajā notebook izmantoju Titanic datu kopu, lai saprastu, ko tieši šeit vajadzētu prognozēt un kādi dati tam ir pieejami.

In [1]:
import seaborn as sns

df = sns.load_dataset('titanic')

print(f"Rindu skaits: {df.shape[0]}")
print(f"Kolonnu skaits: {df.shape[1]}")
df.head()


Rindu skaits: 891
Kolonnu skaits: 15


,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,deck,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,NaN,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,C,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,NaN,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,C,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,NaN,Southampton,no,True


## A. Problēmas apraksts

Šajā datu kopā gribu prognozēt, kādi dati norāda uz to, kas pasažieriem palīdzēja izdzīvot vai nē. To var mēģināt noteikt pēc tādām pazīmēm kā vecums, dzimums, biļetes klase un citiem datiem par pasažieriem. Manuprāt, šī problēma ir svarīga, jo tā palīdz saprast, kuri faktori visvairāk bija tieši saistīti ar izdzīvošanu. Šāds rezultāts būtu noderīgs analītiķim, kurš grib izvērtēt drošības riskus un pieņemt lēmumus, kuri ir balstīti uz datiem.


## B. ML uzdevuma tips

Es šo uzdevumu klasificēju kā **supervised learning**, jo datu kopā jau ir dota pareizā atbilde - kolonna `survived`. Tā ir **klasifikācijas** problēma, jo jāparedz viena no divām klasēm: `0` vai `1`, nevis kāda nepārtraukta skaitliska vērtība.


## C. Target variable

Target mainīgais šeit ir `survived`. Zemāk parādu, cik pasažieru izdzīvoja un cik neizdzīvoja, kā arī kopējo izdzīvojušo procentu.


In [2]:
print(df['survived'].value_counts())
print(f'\nIzdzīvoja: {df["survived"].mean():.1%}')


survived
0    549
1    342
Name: count, dtype: int64

Izdzīvoja: 38.4%


Kolonna `survived` ir tā vērtība, kuru modelis mēģinātu paredzēt. `0` nozīmē, ka pasažieris neizdzīvoja, bet `1` nozīmē, ka pasažieris izdzīvoja.


## D. Features saraksts

Zemāk ir redzams features pārskats, sadalot tās skaitliskajās un kategoriskajās pazīmēs.


In [3]:
print('Skaitliskās features:')
print([col for col in df.select_dtypes(include='number').columns.tolist() if col != 'survived'])
print('\nKategoriskās features:')
print(df.select_dtypes(exclude='number').columns.tolist())


Skaitliskās features:
['pclass', 'age', 'sibsp', 'parch', 'fare']

Kategoriskās features:
['sex', 'embarked', 'class', 'who', 'adult_male', 'deck', 'embark_town', 'alive', 'alone']


## E. Sākotnējie novērojumi

Redzams, ka datu kopā ir gan skaitliskās, gan kategoriskās kolonnas, tāpēc vēlāk modelēšanā būtu jāprot strādāt ar dažādiem datu tipiem. Izdzīvoja apmēram 38.4% pasažieru, tātad lielākā daļa pasažieru neizdzīvoja. Vēl pamanīju, ka kolonna `alive` būtībā atkārto `survived` tekstuālā formā, tāpēc to nevajadzētu izmantot kā ievades pazīmi.
